### DISCUSSION TITLE
Google Takeout and photos added by other contributors to an album

### DISCUSSION THREAD

**Discussion by asmartin** [simulot/immich-go#71] [Q&A]
I'm using Google Takeout to create an export of everything in my Google Photos account and import with `immich-go` which is working great! However, I am not sure I fully understand how the following situations work:

- an album I created with a subset of the photos uploaded by others
- an album shared with me where I have added some photos

In both of these cases, are these complete albums included in my Google Takeout export? In the first case, I see the album I own but the number of photos is substantially less than what I see in Google Photos, so I am guessing the photos I see in Immich are the ones _I added_ to the album, but the ones others added are missing. Does this mean that for all albums I have created in Google Photos where others are collaborators or albums shared with me, I need to also download the album manually in addition to using my Google Takeout export to ensure I get a copy of everything?

Thanks!

**Comment by simulot** [comment]
Thank you for your nice comment!

I'm not sure is shared albums contains only your photos or all... Usually, I'm the serial photographier, so my pictures are their anyway.

You can check in the takeout archive. Open the album folder, and search for json files. If some pictures are missing, you can download the album from the web site and import the result into an immich album `-album "ALBUM NAME"`. But the quality will be inferior and the date of capture will miss.

**Comment by asmartin** [comment]
Thanks for the guidance. My takeout archive is split into several files actually, but I opened up each of them and extracted this album into a single folder on my machine. After doing that, I can see that there are in total (excluding json files) around 1500 images, however the album on Google Photos shows over 3500 images. Next I opened the album on Google Photos and clicked `Download All`. Once this completed, I checked the number of files in the archive and sure enough, it pretty much matches what I see for the total file count in Google Photos. I then imported this zip export using `-album MyAlbum`, but when I refresh the album in Immich the total is well below the total in Google Photos. I ran it again today (same exact command) and I see at the end the following:
```
Managing albums
Update the album MyAlbum
1251 asset(s) added to the album "MyAlbum"
3521 media scanned, 0 uploaded.
```

I reviewed the output and no new files were uploaded, so they were all already present in Immich, but just not properly associated with the album.

Based on this experience, I think the following are good guidelines:

- if you have a shared album, even if you own it, Google Takeout will not export photos that others have added, so make sure to `Download All` on the album as well as using Google Takeout
- if someone else shared an album with you, it will not be included in your Google Takeout and you need to `Download All` in it as well
- even if `immich-go` reports a success on uploading all files to Immich, it may be worth re-running it a couple of times to make sure they are correctly associated with an album, especially when dealing with a large number of images

> But the quality will be inferior and the date of capture will miss.

What makes the quality inferior with this method? Isn't the image file resolution the same as it would have been otherwise? Also, as far as I can tell, the date of capture is importing correctly (it looks like the EXIF data is still present on the images).

Thanks again for the help!

**Comment by simulot** [comment]
Great to see how you have managed it.

When downloading a album from the google photo album's page you get only files without EXIF nor json file. Hence my remark on the dates.

**Comment by simulot** [comment]
> What makes the quality inferior with this method? Isn't the image file resolution the same as it would have been otherwise? Also, as far as I can tell, the date of capture is importing correctly (it looks like the EXIF data is still present on the images).

easy: it keeps the biggest file that is supposed to be the original

**Reply by asmartin** [reply]
That makes sense. If indeed `Download All` does not grab the original file, then at least importing the Google Takeout archive first will ensure you have the original file for images you uploaded and then importing the `Download All` archive after will fill in the gaps of any images others added

**Reply by simulot** [reply]
Yes.

Don't forget about the possibility you have with google photos to store your file but compressed.

**Reply by asmartin** [reply]
True, but I don't think we have any options other than the two already listed so this is as good as it gets
